.github/workflows/s3_connect.yml

```diff
@@ -0,0 +1,87 @@
+# adapted GA workflow from https://github.com/stackhpc/s3-active-storage-rs
+---
+name: S3 Test Connect
+
+on:
+  push:
+    branches:
+      - main
+      - add_s3_tests
+  schedule:
+    - cron: '0 0 * * *'  # nightly
+
+jobs:
+  linux-test:
+    runs-on: "ubuntu-latest"
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10", "3.11"]
+      fail-fast: false
+    name: Linux Python ${{ matrix.python-version }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: activestorage
+          environment-file: environment.yml
+          python-version: ${{ matrix.python-version }}
+          miniforge-version: "latest"
+          miniforge-variant: Mambaforge
+          use-mamba: true
+      - shell: bash -l {0}
+        run: conda --version
+      - shell: bash -l {0}
+        run: python -V
+
+      - name: Configure compliance test suite
+        run: |
+          echo 'PROXY_URL = "http://localhost:8080"' >> config.py
+
+      - name: Start minio object storage
+        run: |
+          s3_scripts/minio-start
+
+      - name: Wait for minio object storage to start
+        run: |
+          until curl -if http://localhost:9001; do
+            sleep 1;
+          done
+
+      - name: Run container
+        run: docker run -it --detach --rm --net=host --name s3-active-storage ghcr.io/stackhpc/s3-active-storage-rs:latest
+
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: activestorage-minio
+          environment-file: environment.yml
+          python-version: ${{ matrix.python-version }}
+          miniforge-version: "latest"
+          miniforge-variant: Mambaforge
+          use-mamba: true
+      - name: Put data in AWS bucket
+        shell: bash -l {0}
+        run: |
+          aws s3 mb s3://pyactivestorage-17666C4471777760
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: 'eu-west-1'
```

Comment:
@markgoddard can I ask for a bit of help please, mate? How/what credentials should I declare to build an AWS bucket in Minio, and then how can I plop data in there? Sorry, am a total Amazon n00b (AWS, otherwise a pro Prime member :laughing: ). Oh and apologies if you are following this PR, hope you didn't get bombarded with notification emails, I had to work straight on Github Actions since I need to update the OS on me work computer, the poor thing's running an ancient Linux, and `docker` stopped working around 1987 on it :grin: Attn: @bnlawrence too, so we learn AWS magic

Reply:
Hi @valeriupredoi, I'm not sure exactly what you're trying to do, but the way I set up the tests it writes the necessary data to the bucket at the beginning of each test case. See https://github.com/valeriupredoi/PyActiveStorage/pull/77/files#diff-ec38c61409198c1e08081e821022002e54d2c0e5d3db3f03787cc9472d926bceR90.

So the github workflow should not need to write any data to S3 directly.

Reply:
cheers, Mark! I'll plug that functionality in the test that's failing - that test that is failing is now run on Minio so the bucket creation should work without any actual amazon credentials right?

Reply:
I'm away tomorrow and next week, but @sd109 should be able to help you.

Reply:
cheers, enjoy your hols! I'll try fix things as much as I can (I need to learn this AWS new set of tricks myself hehe), if not I'll ping Scott :beer: